sdk/python/kfp/client/client.py

```diff
@@ -495,16 +491,16 @@ def get_pipeline_id(self, name: str) -> Optional[str]:
         """
         pipeline_filter = json.dumps({
             'predicates': [{
-                'op': _FILTER_OPERATIONS['EQUALS'],
-                'key': 'name',
+                'operation': _FILTER_OPERATIONS['EQUALS'],
```

Comment:
Would the key change from `'op'` to `'operation'` be a breaking change?

Reply:
Unfortunately yes, this is a breaking change. I guess when we made the API changes, we missed that SDK users could pass in this as a string. 
Technically we could avoid breaking SDK users by loading the JSON back to a dict and do a string replacement on behalf of the user, but I feel a bit hesitate to introduce such a hack that alters user provided value. Hopefully this breakage would be easy to discover and easy to fix. WDYT?

Reply:
Agreed that this is a very minor breaking change. I'm okay with not fixing. It will be easy to add that hack if it turns out to be a pain point for users. Plus, we don't promise backward compatibility from the main namespace to KFP SDK v2 anyway and the client is associate most with the main namespace in KFP v1.